jupyterhub/playbook.yml

```diff
@@ -0,0 +1,214 @@
+---
+# Playbook for maintaining fallback public jupyterhub training server
+# Ensure https_letsencrypt_enabled is set to True in production to
+# ensure certbot handles the letsencrypt certificate setup
+
+- hosts: ome-training-jh.openmicroscopy.org
+
+  roles:
+
+  - role: openmicroscopy.docker
+    docker_additional_options:
+      # WARNING: This disables the default docker network forwarding rules
+      # so we can restrict outgoinng connections from JupyterHub servers,
+      # but this means we must manually configure everything
+      iptables: False
+
+
+- hosts: ome-training-jh.openmicroscopy.org
+  # Separate playbook to force refresh of facts (including docker network
+  # interface)
+  pre_tasks:
+
+  - name: Refresh facts
+    setup:
+
+  roles:
+
+  - role: idr.idr-jupyter
+    idr_jupyter_prefix: /training
+    idr_jupyter_authenticator: tmpauthenticator.TmpAuthenticator
+    idr_jupyter_cull_options:
+    - --timeout=300
+    - --max-age=86400
+    - --cull-users
+    idr_jupyter_hub_log_level: INFO
+    # Allow any username
+    idr_jupyter_users: []
+    # No admins
+    idr_jupyter_admins: []
+    #idr_jupyter_pull_latest: True
+    idr_jupyter_notebook_image: "{{ idr_jupyter_notebook_image_override | default('openmicroscopy/training-notebooks:0.4.0') }}"
```

Comment:
When do you choose to use `default()` and when `defaults/main.yml` ?

Reply:
`idr_jupyter_notebook_image_override` is only used in the molecule test to avoid downloading a massive image: https://github.com/openmicroscopy/prod-playbooks/pull/93/files#diff-0c13968c4ad5f3d4212f08d3d3154da6R43